wp/lib/bap_wp/src/environment.mli

```diff
@@ -201,6 +212,27 @@ val is_x86 : Bap.Std.Arch.t -> bool
     when generating symbols in the function specs at a function call site. *)
 val use_input_regs : t -> bool
 
+(** Obtains a Z3 expression that checks if an address is within the stack region
+    of memory: i.e. `stack_min <= addr <= stack_max`. *)
+val in_stack : t -> Constr.z3_expr -> Constr.z3_expr
+
+(** Obtains a Z3 expression that checks if an address is within the heap region
+    of memory: i.e. `heap_min <= addr <= heap_max`. *)
+val in_heap : t -> Constr.z3_expr -> Constr.z3_expr
+
+(** [mk_init_var env var suffix] creates a fresh Z3 variable that represents the
+    initial state of variable [var] with suffix 'orig' for the original binary
+    and 'mod' for the modified binary. *)
+val mk_init_var : t -> Bap.Std.Var.t -> string -> Constr.z3_expr
+
+(** Add a new binding to the environment for a bap variable to a Z3 expression
+    that represents its initial state. *)
+val set_init_var : t -> Bap.Std.Var.t -> Constr.z3_expr -> t
```

Comment:
Do we need both `mk_init_var` and `set_init_var`? Do we ever call one without the other?